.github/workflows/rust.yml

```diff
@@ -0,0 +1,28 @@
+name: Rust CI
+on:
+  pull_request:
+    paths:
+      - 'rust/**'
+  push:
+    branches:
+      - master
+    paths:
+      - 'rust/**'
```

Comment:
If we do more mono-repo stuff, we might want to limit CI to the parts that changes (apart from bigger integration tests).